gradle/setup.gradle

```diff
@@ -14,107 +14,118 @@
-			description = gradleProject.description
-			url = 'https://github.com/reactor/reactor-core'
-			organization {
-				name = 'reactor'
-				url = 'https://github.com/reactor'
-			}
-			licenses {
-				license {
-					name 'Apache License, Version 2.0'
-					url 'https://www.apache.org/licenses/LICENSE-2.0.txt'
-					distribution 'repo'
+			pom {
+				afterEvaluate {
+					name = project.description
+					description = project.description
 				}
-			}
-			scm {
+
 				url = 'https://github.com/reactor/reactor-core'
-				connection = 'scm:git:git://github.com/reactor/reactor-core'
-				developerConnection = 'scm:git:git://github.com/reactor/reactor-core'
-			}
-			developers {
-				developer {
-					id = 'smaldini'
-					name = 'Stephane Maldini'
-					email = 'smaldini@pivotal.io'
+				organization {
+					name = 'reactor'
+					url = 'https://github.com/reactor'
 				}
-				developer {
-					id = 'akarnokd'
-					name = 'David Karnok'
-					email = 'akarnokd@gmail.com'
+				licenses {
+					license {
+						name = 'Apache License, Version 2.0'
+						url = 'https://www.apache.org/licenses/LICENSE-2.0.txt'
+						distribution = 'repo'
+					}
 				}
-				developer {
-					id = 'simonbasle'
-					name = 'Simon Baslé'
-					email = 'sbasle@pivotal.io'
+				scm {
+					url = 'https://github.com/reactor/reactor-core'
+					connection = 'scm:git:git://github.com/reactor/reactor-core'
+					developerConnection = 'scm:git:git://github.com/reactor/reactor-core'
 				}
-				developer {
-					id = 'rsivaram'
-					name = 'Rajini Sivaram'
-					email = 'rsivaram@pivotal.io'
+				developers {
+					developer {
+						id = 'smaldini'
+						name = 'Stephane Maldini'
+						email = 'smaldini@pivotal.io'
+					}
+					developer {
+						id = 'akarnokd'
+						name = 'David Karnok'
+						email = 'akarnokd@gmail.com'
+					}
+					developer {
+						id = 'simonbasle'
+						name = 'Simon Baslé'
+						email = 'sbasle@pivotal.io'
+					}
+					developer {
+						id = 'bsideup'
+						name = 'Sergei Egorov'
+						email = 'segorov@pivotal.io'
+					}
+					developer {
+						id = 'sdeleuze'
+						name = 'Sébastien Deleuze'
+						email = 'sdeleuze@pivotal.io'
+					}
 				}
-				developer {
-					id = 'sdeleuze'
-					name = 'Sébastien Deleuze'
-					email = 'sdeleuze@pivotal.io'
+				issueManagement {
+					system = "GitHub Issues"
+					url = "https://github.com/reactor/reactor-core/issues"
+				}
+
+				withXml {
+					//set optional true for compileOnly dependencies, which are not in there by default
+					asNode().dependencies[0].each { node ->
+						def group = node.groupId.text()
+						def artifact = node.artifactId.text()
+						def version = node.version.text()
+						def isOptional = project.configurations.optional.allDependencies.any { dep -> dep.group == group && dep.name == artifact }
+						if (isOptional) {
+							node.appendNode('optional', true)
+							println "$group:$artifact:$version has been marked as optional: " + node.text()
```

Comment:
```suggestion
							println "$group:$artifact:$version has been marked as optional: ${node.text()}"
```

Reply:
actually, we don't even really need the full node text (was there for debugging)